docs/site/Testing-your-application.md

```diff
@@ -408,17 +408,68 @@ Verify how was the stubbed method executed at the end of your unit test (in the
 ```ts
 // expect that repository.find() was called with the first
 // argument deeply-equal to the provided object
-sinon.assert.calledWithMatch({where: {id: 1}});
+sinon.assert.calledWithMatch(findStub, {where: {id: 1}});
 ```
 
 See [Unit test your controllers](#unit-test-your-controllers) for a full
 example.
 
 #### Create a stub Service
 
-{% include content/tbd.html %}
+If your controller relies on service proxy for service oriented backends such as
+REST APIs, SOAP Web Services, or gRPC microservices, then we can create stubs
+for the service akin to the steps outlined in the above
+[stub repository](#create-a-stub-repository) section. Consider a dependency on a
+`GeoCoder` service that relies on a remote REST API for returning coordinates
+for a specific address.
 
-The initial beta release does not include Services as a first-class feature.
+```ts
+class DummyGeocoder implements GeocoderService {
+  // in practice, the constructor could take an API key to authenticate with
+  // the service proxy provider
+  constructor() {}
+
+  geocode(address: string) {
+    // this would call the remote API for the real coordinates
+  }
+}
+```
+
+Create a test double for the service class:
+
+```ts
+describe('GeoCoderController', () => {
+  let geoService: DummyGeocoder;
+  beforeEach(givenStubbedService);
+
+  // your unit tests
+
+  function givenStubbedService() {
+    geoService = sinon.createStubInstance(DummyGeocoder);
+  }
+});
+```
+
+Configure the `geocode` method as a stub and its behaviour before your test(s):
+
+```ts
+// geoService.geocode() will return a promise that
+// will be resolved with the provided array
+const geocodeStub = geoService.geocode as sinon.SinonStub;
+geocodeStub.resolves([<GeoPoint>{y: 41.109653, x: -73.72467}]);
+```
+
+Verify how the stub was executed:
+
+```ts
+// expect that geoService.geocode() was called with the first
+// argument equal to the provided address string
+sinon.assert.calledWithMatch(geocodeStub, '1 New Orchard Road, Armonk, 10504');
+```
+
+Check out
+[this](<(https://github.com/strongloop/loopback-next/blob/master/examples/todo/src/__tests__/unit/controllers/todo.controller.unit.ts#L53-L71)>)
```

Comment:
From the github view, the `this` doesn't show the link.  do you mean:
```
[this](https://github.com/strongloop/loopback-next/blob/master/examples/todo/src/__tests__/unit/controllers/todo.controller.unit.ts#L53-L71)
```


Also, I'm not sure specifying the line number would be a good solution in general, because they tend to get outdated easily.  But I'm not sure if specifying the test case name would be a good alternative either.  Just wanna bring this up.  Thanks.

Reply:
1) Let's use a more descriptive link name than "this". For example, "TodoController unit tests".

2) Either point to the entire file in `master` branch (`https://github.com/strongloop/loopback-next/blob/master/examples/todo/src/__tests__/unit/controllers/todo.controller.unit.ts`)  or lock down the commit sha in a link pointing to line numbers (`https://github.com/strongloop/loopback-next/blob/bd0c45033503f631a533ad6176620354d9cf6768/examples/todo/src/__tests__/unit/controllers/todo.controller.unit.ts#L53-L71`).

Reply:
Yes I was looking for the commit sha, thank you both for pointing this out :-).